prov/gni/test/mr.c

```diff
@@ -76,6 +79,22 @@ uint64_t default_flags = 0;
 uint64_t default_req_key = 0;
 uint64_t default_offset = 0;
 
+struct timeval s1, s2;
+
+static void calculate_time_difference(struct timeval *start, struct timeval *end,
+		int *secs_out, int *usec_out)
+{
+	*secs_out = end->tv_sec - start->tv_sec;
+	if (end->tv_usec < start->tv_usec) {
+		*secs_out = *secs_out - 1;
+		*usec_out = (1000000 + end->tv_usec) - start->tv_usec;
+	} else {
+		*usec_out = end->tv_usec - start->tv_usec;
+	}
+}
+
+
+
```

Comment:
Is this the same as the one in bitmap?  If so, maybe we should move it to a file of shared functions?